### PR TITLE
Docs for AspNet Instrumentation

### PR DESCRIPTION
**Pull request by cijothomas**
Release day = Docs day. Lot more to come!

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNet/README.md [hunk 1]
```diff
@@ -36,30 +36,61 @@ Configuration with ASP.NET (Full .NET Framework) running in IIS or IIS Express
...
+filtering of requests by using `Filter` function in `AspNetInstrumentationOptions`.
+This can be used to filter out any requests based on some condition. The Filter
+receives the `HttpContext` for the request, and filters out the request if the Filter
+returns false or throws exception.
+
+The following shows an example of Filter being used to filter out all POST requests.
+
+```csharp
+this.tracerProvider = Sdk.CreateTracerProviderBuilder()
+                 .AddAspNetInstrumentation(
```
Comment (eddynaka) [inline src/OpenTelemetry.Instrumentation.AspNet/README.md:76]: nit: check the identation

File: examples/Console/TestConsoleExporter.cs [hunk 2]
```diff
@@ -102,16 +102,13 @@ internal class MyProcessor : ActivityProcessor
...
-                    }
-                    else if (activity.Kind == ActivityKind.Client)
-                    {
-                        activity.SetTag("customClientTag", "Custom Tag Value for Client");
-                    }
+                    activity.SetTag("customServerTag", "Custom Tag Value for server");
+                }
+                else if (activity.Kind == ActivityKind.Client)
+                {
+                    activity.SetTag("customClientTag", "Custom Tag Value for Client");
```
Comment (alanwest) [inline examples/Console/TestConsoleExporter.cs:111]: Minor thing. A little goofy that `Client` is caps and `server` is not.

File: examples/AspNet/Global.asax.cs [hunk 3]
```diff
@@ -32,9 +32,41 @@ public class WebApiApplication : HttpApplication
...
+                        {
+                            if (httpContext.Request.HttpMethod.Equals("POST"))
+                            {
+                                return false;
+                            }
+                            else
+                            {
+                                return true;
+                            }
+                        };
```
Comment (reyang) [inline examples/AspNet/Global.asax.cs:49]:
```suggestion
                        options.Filter = (httpContext) => !httpContext.Request.HttpMethod.Equals("POST");
```

File: examples/AspNet/Global.asax.cs [hunk 4]
```diff
@@ -32,9 +32,41 @@ public class WebApiApplication : HttpApplication
...
         {
+            this.tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddAspNetInstrumentation(
+                    (options) =>
+                    {
+                        options.Filter = (httpContext) =>
+                        {
+                            if (httpContext.Request.HttpMethod.Equals("POST"))
+                            {
+                                return false;
```
Comment (ThomsonTan) [inline examples/AspNet/Global.asax.cs:43]: Comment on what `true/false` mean?

File: docs/trace/building-your-own-processor/MyEnrichmentActivityProcessor.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+// </copyright>
+
+using System.Diagnostics;
+using OpenTelemetry.Trace;
+
+internal class MyEnrichmentActivityProcessor : ActivityProcessor
+{
+    public override void OnStart(Activity activity)
+    {
+        activity.SetTag("MyAdditionalTag", "AdditionalTagValue");
```
[the remark this answers is not in the document]
Comment (giulianob) [reply docs/trace/building-your-own-processor/MyEnrichmentActivityProcessor.cs:24]: As an outsider who just started using this library, also explaining difference between tag and baggage would be helpful. Show adding both and what the difference is. Just ran into this and went into the official spec to go learn about them.

**Comment by cijothomas** [comment]
I will revert changes to Processor part as this PR is making processors specific to the use case of Enrichment.
Will use this PR for aspnet changes only.
for processor as an enrichment - i'll expand getting started guide to include more of this basic scenarios.

File: docs/trace/building-your-own-processor/MyEnrichmentActivityProcessor.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (cijothomas) [reply docs/trace/building-your-own-processor/MyEnrichmentActivityProcessor.cs]: This section is intended to be the doc for Processors. Explaining Activity tag/baggage would go elsewhere.

File: docs/trace/building-your-own-processor/MyActivityProcessor.cs [hunk 6]
```diff
@@ -1,50 +1,50 @@
...
-        Console.WriteLine($"{this}.OnForceFlush");
-        return true;
-    }
-
-    protected override void OnShutdown(int timeoutMilliseconds)
-    {
-        Console.WriteLine($"{this}.OnShutdown");
-    }
-}
+﻿// <copyright file="MyActivityProcessor.cs" company="OpenTelemetry Authors">
```
Comment (reyang) [inline docs/trace/building-your-own-processor/MyActivityProcessor.cs:1]: Is there a CRLF/LF change? We can probably do all the normalization in one shot.

Comment (cijothomas) [reply docs/trace/building-your-own-processor/MyActivityProcessor.cs:1]: yes. I am abandoning this PR anyway in favor of 1205

**Pull request closed without merging**